### PR TITLE
Do not try to strartup dropped replica

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Upgrade from 20.3 and older to 22.9 and newer should be done through an intermediate version if there are any `ReplicatedMergeTree` tables, otherwise server with new version will not start. 

Related to: https://github.com/ClickHouse/ClickHouse/pull/40148#discussion_r949518408, https://github.com/ClickHouse/ClickHouse/pull/30826#issuecomment-1220610613

### CODE REVIEW

**Comment by tavplubix** [comment]
Integration tests failures are expected:
`test_version_update_after_mutation/test.py::test_upgrade_while_mutation`
`test_version_update_after_mutation/test.py::test_mutate_and_upgrade`
`test_no_local_metadata_node/test.py::test_table_start_without_metadata`

These tests check compatibility between 20.4 and older versions. I think we can safely remove `test_no_local_metadata_node` and change tag to newer in `test_version_update_after_mutation` (it may be useful to have a test that runs mutation during upgrade).
cc: @alesapin

File: src/Storages/MergeTree/ReplicatedMergeTreeAttachThread.cpp [hunk 1]
```diff
@@ -77,6 +82,23 @@ void ReplicatedMergeTreeAttachThread::run()
     }
 }
 
+void ReplicatedMergeTreeAttachThread::checkHasReplicaMetadataInZooKeeper(const zkutil::ZooKeeperPtr & zookeeper, const String & replica_path)
+{
+    /// Since 20.4 and until 22.9 "/metadata" and "/metadata_version" nodes were created on replica startup.
+    /// Since 21.12 we could use "/metadata" to check if replica is dropped (see StorageReplicatedMergeTree::dropReplica),
```
Comment (antonio2368) [inline src/Storages/MergeTree/ReplicatedMergeTreeAttachThread.cpp:88]:
```suggestion
    /// Until 21.12 we could use "/metadata" to check if replica is dropped (see StorageReplicatedMergeTree::dropReplica),
```
?

**Review by antonio2368 — APPROVED** [review]

File: src/Storages/MergeTree/ReplicatedMergeTreeAttachThread.cpp (hunk 1 above)
Comment (tavplubix) [reply src/Storages/MergeTree/ReplicatedMergeTreeAttachThread.cpp:88]: No, I mean that code that removes `/metadata` first before removing other nodes was added in 21.12

**Comment by tavplubix** [comment]
Stress test (debug) - https://github.com/ClickHouse/ClickHouse/issues/40399

**Pull request merged**